src/Tags.elm

```diff
@@ -0,0 +1,217 @@
+module Tags exposing (commonTagsRule, expressionTagsRule, ruleConfig)
+
+import Elm.Syntax.Expression exposing (Expression(..), LetDeclaration(..))
+import Elm.Syntax.Node as Node exposing (Node(..))
+import ElmSyntaxHelpers
+import Json.Encode as Encode exposing (Value)
+import Review.ModuleNameLookupTable as LookupTable exposing (ModuleNameLookupTable)
+import Review.Rule as Rule exposing (Rule)
+import RuleConfig exposing (AnalyzerRule(..), RuleConfig)
+import Set exposing (Set)
+
+
+type alias Tag =
+    String
+
+
+type alias ProjectContext =
+    Set Tag
+
+
+type alias ModuleContext =
+    { lookupTable : ModuleNameLookupTable
+    , tags : Set Tag
+    }
+
+
+ruleConfig : RuleConfig
+ruleConfig =
+    { restrictToFiles = Nothing
```

Comment:
Will this include any template type files that are provided along with the solution?

Reply:
Yes, it will, that's a good point, the data extractor will run on the whole project (it can even look at the `elm.json` if we want). 
Do we want to skip the support files? If we do, we should enforce a convention on their names (I think we already do that, but it should be in the CI).

@ErikSchierboom what does the C# tagger do when it comes to solutions with `editor` files in it? Skip it? Include it?

You just made me realize that this will probably  tag the tests as well, which I guess we don't want, so I'll try to avoid that.

Adding file names in the field `restrictToFiles` won't work though, because it's eventually fed `Rule.filterErrorsForFiles` which I _think_ only works for errors, not for data extractions. I should rename it to reflect that.